jaggery-modules/rxt/module/scripts/core/core.js

```diff
@@ -625,6 +625,40 @@ var core = {};
         }
         return false;
     };
+    RxtManager.prototype.isSolarFacetsEnabled = function(type){
+        var rxtDefinition = this.rxtMap[type];
+        if (!rxtDefinition) {
+            log.error('Unable to locate the rxt definition for type: ' + type);
+            throw 'Unable to locate the rxt definition for type: ' + type
+            + ' in order to determine if the asset can be downloaded';
+        }
+        if ((rxtDefinition.meta) && (rxtDefinition.meta.categorization)) {
+            return rxtDefinition.meta.categorization.solarFacetsEnabled || false;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug('Unable to locate the  meta property to determine whether ' +
+                'solar facets is enabled in categorization for ' + type
+                + '.Make sure the meta property is present in the configuration callback of the asset.js');
+        }
+        return false;
+    };
+    RxtManager.prototype.collapseInCount = function(type){
```

Comment:
Please refer to my earlier comment about renaming this method.

Please add comments